garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,257 @@ local meta = FindMetaTable( "Vector" )
 -- Nothing in here, still leaving this file here just in case
 
 --[[---------------------------------------------------------
-	Converts Vector To Color - alpha precision lost, must reset
+Converts Vector to Color - alpha precision lost, must reset
 -----------------------------------------------------------]]
-function meta:ToColor( )
+function meta:ToColor(num)
+	local a = (num == nil and 1 or num)
+	local x, y, z = self:Unpack()
+	return Color(x * 255, y * 255, z * 255, a * 255)
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Table - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToTable()
+	local x, y, z = self:Unpack()
+	return {x = x, y = y, z = z}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Array - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToArray()
+	return {self:Unpack()}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Angle - data is copied
+-----------------------------------------------------------]]
+function meta:ToAngle()
+	return Angle(self:Unpack())
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a rotated vector. Kind of like v:GetNormalized()
+-----------------------------------------------------------]]
+function meta:GetRotated(...)
+	local v = Vector(self)
+	v:Rotate(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(...)
+	local v = Vector(self)
+	v:Mul(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(...)
+	local v = Vector(self)
+	v:Add(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a vector
+-----------------------------------------------------------]]
+function meta:AddUnpacked(x, y, z)
+	local vx, vy, vz = self:Unpack()
+	self:AddUnpacked(vx + x, vy + y, vz + z)

```

Comment:
I think you meant to do SetUnpacked here, this is infinite recursion right now.

Reply:
Yes ! Thanks !